src/background_thread.c

```diff
@@ -850,7 +849,8 @@ background_thread_boot1(tsdn_t *tsdn, base_t *base) {
 		background_thread_info_init(tsdn, info);
 		malloc_mutex_unlock(tsdn, &info->mtx);
 	}
+	/* Using _impl to bypass the locking check during init. */
+	background_thread_enabled_set_impl(opt_background_thread);
```

Comment:
The initialization above (all the background thread relevant inits) is free from racy conditions and putting this setting before or after the initialization is fine, is that correct?

Reply:
Right. All the `_boot()` functions are covered by the malloc_init lock.

Reply:
(but I moved it to the end of the function, because there are still error exit paths in the middle, and technically we should not mark it as enabled unless all init succeeded)